src/runtime/TemporalObject.h

```diff
@@ -351,6 +352,128 @@ class TemporalZonedDateTime : public Temporal {
+    static bool isValidDuration(const std::array<int, 10>& fields);
+    static int durationSign(const std::array<int, 10>& fields);
+    static std::map<std::string, int> createDurationRecord(ExecutionState& state, int years, int months, int weeks, int days, int hours, int minutes, int seconds, int milliseconds, int microseconds, int nanoseconds);
+    static Value createTemporalDuration(ExecutionState& state, int years, int months, int weeks, int days, int hours, int minutes, int seconds, int milliseconds, int microseconds, int nanoseconds, Optional<Object*> newTarget);
+
+    static Value toTemporalDuration(ExecutionState& state, const Value& item);
+    static std::map<std::string, int> toTemporalDurationRecord(ExecutionState& state, const Value& temporalDurationLike);
+    static std::map<std::string, Value> toTemporalPartialDurationRecord(ExecutionState& state, const Value& temporalDurationLike);
+    static Value createNegatedTemporalDuration(ExecutionState& state, const Value& duration);
+
+    bool isTemporalDurationObject() const override
+    {
+        return true;
+    }
+
+    int getYear() const
+    {
+        return m_year;
+    }
+    void setYear(int mYear)
+    {
+        m_year = mYear;
+    }
+    int getMonth() const
+    {
+        return m_month;
+    }
+    void setMonth(int mMonth)
+    {
+        m_month = mMonth;
+    }
+    int getWeek() const
+    {
+        return m_week;
+    }
+    void setWeek(int mWeek)
+    {
+        m_week = mWeek;
+    }
+    int getDay() const
+    {
+        return m_day;
+    }
+    void setDay(int mDay)
+    {
+        m_day = mDay;
+    }
+    int getHour() const
+    {
+        return m_hour;
+    }
+    void setHour(int mHour)
+    {
+        m_hour = mHour;
+    }
+    int getMinute() const
+    {
+        return m_minute;
+    }
+    void setMinute(int mMinute)
+    {
+        m_minute = mMinute;
+    }
+    int getSecond() const
+    {
+        return m_second;
+    }
+    void setSecond(int mSecond)
+    {
+        m_second = mSecond;
+    }
+    int getMillisecond() const
+    {
+        return m_millisecond;
+    }
+    void setMillisecond(int mMillisecond)
+    {
+        m_millisecond = mMillisecond;
+    }
+    int getMicrosecond() const
+    {
+        return m_microsecond;
+    }
+    void setMicrosecond(int mMicrosecond)
+    {
+        m_microsecond = mMicrosecond;
+    }
+    int getNanosecond() const
+    {
+        return m_nanosecond;
+    }
+    void setNanosecond(int mNanosecond)
+    {
+        m_nanosecond = mNanosecond;
+    }
+    TemporalCalendar* getCalendar() const
+    {
+        return m_calendar;
+    }
+    void setCalendar(TemporalCalendar* mCalendar)
+    {
+        m_calendar = mCalendar;
+    }
+
+private:
+    int m_year;
+    int m_month;
+    int m_week;
+    int m_day;
+    int m_hour;
+    int m_minute;
+    int m_second;
+    int m_millisecond;
+    int m_microsecond;
+    int m_nanosecond;
+    TemporalCalendar* m_calendar;
```

Comment:
These members are defined in other class like `TemporalPlainTime`.
How about assigning an intermediate class that defines these members to reduce code duplication?